ql/src/java/org/apache/hadoop/hive/ql/udf/UDFToByte.java

```diff
@@ -128,6 +128,11 @@ public ByteWritable evaluate(IntWritable i) {
     if (i == null) {
       return null;
     } else {
+      int value = i.get();
+      if (value < Byte.MIN_VALUE || value > Byte.MAX_VALUE) {
+        throw new IllegalArgumentException("Value out of range for Byte: " + value);
```

Comment:
It is better to use Hive-specific exceptions. Hive has an exception `UDFArgumentException` which seems suitable here.